### DISCUSSION TITLE
Bug in the neovim plugin

### DISCUSSION THREAD

**Discussion by data-stepper** [community/community#16435] [Copilot Conversations] [answered]
So I have been using copilot a lot and have been loving it ever since. I use it mainly to complete python code for me and it works fantastically.
But since the update of the neovim plugin, it doesn't seem to be completing python code anymore. Interestingly it still completes code in other buffers but not in any python buffers. Any fixes on this?

**Comment by szymonmaszke** [comment]
Similar behavior noticed in case of golang, C++/Java/bash/lua seem to work correctly though.

**Comment by Arro** [comment]
Javascript is also not working at the moment.  I tried logging in and out, as well as updating the plugin.   Nothing being suggested.

**Comment by sigmaSd** [comment]
I also want to point out that this problem still persist with the latest version of the plugin 1.3.0 and it seems not to related to the plugin or neovim versions because I downgraded both and the issue persists, maybe its something from copilot server side?

**Comment by data-stepper** [comment]
I think we should open a GitHub issue on the actual neovim plugin, maybe it's a bug in their code and not in Copilot??

**Reply by Arro** [reply]
Alas, you can't open an issue on that repo.  The issues are turned off.  It appears posting here is our only avenue.

**Comment by davecheney** [comment]
If you are using an older version of the plugin, please upgrade to 1.3.0 or later. See https://github.com/github/feedback/discussions/16423

**Reply by Arro** [reply]
Yep, updated to that version.  Also tried rolling back to a previous version.

**Reply by davecheney** [reply]
Looking at your telemetry I see you're still running version 1.12 (I think, the version numbers copilot.vim reports can sometimes be confusing). Please try the mechanism listed here https://github.com/github/feedback/discussions/15684#discussioncomment-2754664

**Reply by Arro** [reply]
Alright so I actually have 2 computers, and I tested it on my other computer and it's still working.  

On the non-working one, I have `cd`d directly into my `~/.vim/bundle/copilot.vim` directory (I use vundle for plugins, which puts things there) and I verified it **is** 1.3.0.  In `autoload/copilot/agent.vim`, it does say `let s:plugin_version = '1.3.0'`.  

But perhaps the issue is that I have two computers?

**Reply by davecheney** [reply]
You can use copilot on as many computers as you would like. Looking at your telemetry it does look like you have two different copilot versions in use, one using version 1.2, the other using 1.3

**Reply by Arro** [reply]
Alright, I upgraded the outdate computer to 1.3.0.  It was the computer that was previously working, a M1 Mac Mini.  

The non-working one is a M1 MacBook Pro.  That one has been on 1.3.0 for multiple days.

It's interesting to me that one computer works for Github Copilot and one doesn't.  

Both computers:
- macOS 12.3.1
- node 16.15.0
- npm 8.5.5
- nvim v0.7.0
- copilot.vim 1.3.0
- identical .vimrc files

So I'm at a loss here what the problem is, but I want to get the laptop working again asap.

**Reply by data-stepper** [reply]
So I just checked and I am also running version 1.3.0 and the issue still persists. I am using vim-plug and it already updated to the latest version but I double checked that by running `git pull` in the repository.
I am also using two machines, both arch linux identical init.vim files and so on and on both, I do get completions for Markdown, LaTeX but not for python.

**Comment by alstjd4153** [comment]
> So I have been using copilot a lot and have been loving it ever since. I use it mainly to complete python code for me and it works fantastically.
> But since the update of the neovim plugin, it doesn't seem to be completing python code anymore. Interestingly it still completes code in other buffers but not in any python buffers. Any fixes on this?

**Reply by davecheney** [reply]
@junhuin please see https://github.com/github/feedback/discussions/16423

**Reply by davecheney** [reply]
There appears to be a bug in Node 18 which breaks some languages.

**Comment by tpope** [comment, marked as answer]
There's currently an issue running on Node.js 18.1.0. Downgrading to an older version is a workaround.  If you can't easily downgrade, you can use a tool like [nodenv](https://github.com/nodenv/nodenv) or [nvm](https://github.com/nvm-sh/nvm) to install a version of Node in your home directory, and then tell Copilot where to find it:

```vim
 let g:copilot_node_command = "~/.nodenv/versions/16.15.0/bin/node"
```

**Reply by data-stepper** [reply]
Yes this solves the issue, on arch one can easily install nodejs version 16.15.0 with
```
sudo pacman -S nodejs-lts-gallium
```
that should also remove version 18 globally.
